src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -440,10 +440,14 @@ public virtual void OnEvaluationStarted(EventArgs e)
         public event EventHandler<EvaluationCompletedEventArgs> EvaluationCompleted;
         public virtual void OnEvaluationCompleted(EvaluationCompletedEventArgs e)
         {
-            this.HasRunWithoutCrash = true;
+            if(e.EvaluationSucceeded)
+            {
+                this.HasRunWithoutCrash = true;
+            }
```

Comment:
I'm not sure what are the odds of `OnEvaluationStarted` not getting called before `OnEvaluationCompleted`. If that ever happens then `HasRunWithoutCrash` will be left with an unknown value. Should we instead do this?

```csharp
// Assignment without condition.
this.HasRunWithoutCrash = e.EvaluationSucceeded;
```